pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -56,6 +56,16 @@ var _ conditions.Setter = &Cluster{}
 // ClusterSpec holds the desired state of the Cluster (from the client).
 type ClusterSpec struct {
 	KubeConfig string `json:"kubeconfig"`
+
+	// Unschedulable controls cluster schedulability of new workloads. By
+	// default, cluster is schedulable.
+	// +optional
+	Unschedulable bool `json:"unschedulable"`
+
+	// Evict controls cluster schedulability of new and existing workloads.
+	// If evict is true, unschedulable is ignored and assumed to be true.
+	// By default, cluster is not evicted.
+	Evict bool `json:"evict"`
```

Comment:
What if we made this EvictAfter so you could set a datetime for it?

Reply:
Sure. That starts to diverge even more from how node eviction works, but we can still change how we want this all to work later.

Seeing an `EvictAfter` in the future on a cluster should enqueue a delayed workqueue item in the future with `AddAfter(duration)`, and `EvictAfter` in the present/past should evict workloads?

Reply:
This is done now. As written currently, this might lead to a bad experience if you set `evictAfter: in ten minutes` without also setting `unschedulable: true` -- a workload might get scheduled to the cluster only to get evicted in ten minutes.

It's not a huge deal -- just cordon it before you schedule it to be drained -- but I wanted to call it out while I still remembered this.

Reply:
If it's easier to test the UX of a bool, happy to start there. @sttts WDYT?

Reply:
You can set it to "time.Now()` to get the boolean behaviour, no? Time looks fine.